### ISSUE TITLE
Newly created jsconfig.json file is not automatically saved

### ISSUE THREAD

**Issue by joaomoreno** [Microsoft/vscode#7359]
- Have `"files.autoSave": "afterDelay"` in your settings
- Open code on a new folder
- Create a `hello.js` file and open it
- Wait for the (ugly) green lightbulb in the bottom right corner to appear; click it and choose to create a `jsconfig.json` file.

The newly created `jsconfig.json` file will not be automatically saved. It will instead display the unsaved icon and wait for me to save it.

**Comment by bpasero** [comment]
As designed I would say, I think @jrieken is opening an untitled file with associated path and this file will not be on disk until you save.

**Comment by jrieken** [comment]
yes - as designed. we don't want to boldly write to you disk, like so you can still discard

**Comment by joaomoreno** [comment]
I gotta reopen this. Here are identical cases which behave as I expect them to:

- Setting up debug `launch.json`;
- Setting up task `tasks.json`;
- Setting up workspace settings.

I clearly want that `jsconfig.json` file, since it took me 2 clicks to get it. If I regret it, it's a delete away. But showing me the unsaved indicator when I have auto-save enabled... 👎

**Comment by bpasero** [comment]
@joaomoreno you talk about auto save here but settings files are being created independent of that setting. Should we turn this around then and argue that any configuration JSON file gets:
* saved to disk automatically if auto save is enabled
* shows up as dirty (and not on disk) if auto save is disabled

?

**Comment by joaomoreno** [comment]
I guess that *turning it around* means, apart from making this case consistent with the others I mentioned, changing the behaviour when auto save is disabled. I am fine with that.

**Comment by bpasero** [comment]
Yeah, that is what I meant.
